From c41f07cf5785b2d755b85f20bf0546c6ce2ebb02 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@redhat.com>
Date: Fri, 16 Aug 2013 17:24:11 +0000
Subject: Add correct flag for reaping the progress child

This fixes the WARNING about ECHILD that comes from some versions
of glib. The WARNING was removed in later versions of glib, but this
is also a good fix.

https://bugzilla.gnome.org/show_bug.cgi?id=697895
---
diff --git a/tool/seahorse-tool-progress.c b/tool/seahorse-tool-progress.c
index 613e82f..c039118 100644
--- a/tool/seahorse-tool-progress.c
+++ b/tool/seahorse-tool-progress.c
@@ -217,7 +217,7 @@ seahorse_tool_progress_start (const gchar *title)
     argv[2] = (gchar *)title;
     argv[3] = NULL;
 
-    ret = g_spawn_async_with_pipes (NULL, argv, NULL, G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_SEARCH_PATH,
+    ret = g_spawn_async_with_pipes (NULL, argv, NULL, G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_SEARCH_PATH | G_SPAWN_DO_NOT_REAP_CHILD,
                                     NULL, NULL, &progress_pid, &progress_fd, NULL, NULL, &err);
 
     if (!ret) {
--
cgit v0.9.2
